base/src/main/java/io/spine/validate/option/Goes.java

```diff
@@ -82,21 +85,19 @@ private boolean canPairedBeRequired(FieldContext context) {
         var field = context.targetDeclaration();
         var messageType = field.declaringType();
         var pairedField = messageType.field(pairedFieldName);
-        var warningMessage = format(
-                "Field `%s` paired with `%s` cannot be checked for presence. " +
-                        "`(goes).with` at %s is obsolete.",
-                pairedField, field, field
-        );
-        return checkType(pairedField, warningMessage);
+        var result = checkType(field);
+        if (!result) {
+            _warn().log(
+                    "Field `%s` paired with `%s` cannot be checked for presence. " +
+                            "`(goes).with` at %s is obsolete.",
+                    pairedField, field, field
+            );
+        }
+        return result;
     }
 
-    private boolean checkType(FieldDeclaration field, String warningMessage) {
+    private static boolean checkType(FieldDeclaration field) {
         var type = field.javaType();
-        if (field.isCollection() || Required.CAN_BE_REQUIRED.contains(type)) {
-            return true;
-        } else {
-            _warn().log(warningMessage);
```

Comment:
That was something we should have retained when addressing the Error Prone warnings.